File: easy/0070_climbing_stairs/solution.py
class Solution:
    def climbStairs(self, n: int) -> int:
        phi = (1 + 5 ** 0.5) / 2
        psi = (1 - 5 ** 0.5) / 2

        return int((phi ** (n + 1) - psi ** (n + 1)) / (phi - psi))

File: easy/0070_climbing_stairs/solution.cpp
class Solution {
public:
  int climbStairs(int n) {
    if (n < 4)
      return n;

    int prev = 2, curr = 3;

    for (int _ = 0; _ < n - 3; ++_) {
      curr += prev;
      prev = curr - prev;
    }

    return curr;
  }
};
